var/spack/repos/builtin/packages/julia/package.py

```diff
@@ -97,162 +93,107 @@ class Julia(Package):
     # USE_SYSTEM_UTF8PROC=0
     # USE_SYSTEM_LIBGIT2=0
 
-    # Run-time dependencies for Julia packages:
-    depends_on("hdf5", when="+hdf5", type="run")
-    depends_on("mpi", when="+mpi", type="run")
-    depends_on("py-matplotlib", when="+plot", type="run")
+    conflicts('@:0.7.0', when='target=aarch64:')
 
-    conflicts("@:0.7.0", when="target=aarch64:")
+    # GCC conflicts
+    conflicts('@:0.5.1', when='%gcc@8:', msg='Julia <= 0.5.1 need GCC <= 7')
+
+    # Building recent versions of Julia with Intel is untested and unsupported
+    # by the Julia project, https://github.com/JuliaLang/julia/issues/23407.
+    conflicts('@0.6:', when='%intel',
+              msg='The Julia project does not currently support building with '
+                  'the Intel compiler.')
+
+    def setup_build_environment(self, env):
+        # The julia build can have trouble with finding GCC libraries with the
+        # spack compiler.
+        if self.compiler.name == 'gcc':
+            gcc_base = os.path.split(os.path.split(self.compiler.cc)[0])[0]
+            env.prepend_path('LD_LIBRARY_PATH', join_path(gcc_base, 'lib64'))
 
     def install(self, spec, prefix):
         # Julia needs git tags
-        if os.path.isfile(".git/shallow"):
-            git = which("git")
-            git("fetch", "--unshallow")
+        if os.path.isfile('.git/shallow'):
+            git = which('git')
+            git('fetch', '--unshallow')
         # Explicitly setting CC, CXX, or FC breaks building libuv, one
         # of Julia's dependencies. This might be a Darwin-specific
         # problem. Given how Spack sets up compilers, Julia should
         # still use Spack's compilers, even if we don't specify them
         # explicitly.
         options = [
-            # "CC=cc",
-            # "CXX=c++",
-            # "FC=fc",
-            # "USE_SYSTEM_ARPACK=1",
-            "override USE_SYSTEM_CURL=1",
-            # "USE_SYSTEM_FFTW=1",
-            # "USE_SYSTEM_GMP=1",
-            # "USE_SYSTEM_MPFR=1",
-            # "USE_SYSTEM_PCRE=1",
-            "prefix=%s" % prefix]
-        if "+cxx" in spec:
-            if "@master" not in spec:
+            # 'CC=cc',
+            # 'CXX=c++',
+            # 'FC=fc',
+            # 'USE_SYSTEM_ARPACK=1',
+            'override USE_SYSTEM_CURL=1',
+            # 'USE_SYSTEM_FFTW=1',
+            # 'USE_SYSTEM_GMP=1',
+            # 'USE_SYSTEM_MPFR=1',
+            # 'USE_SYSTEM_PCRE=1',
+            'prefix={0}'.format(prefix)]
+        if '+cxx' in spec:
+            if '@master' not in spec:
                 raise InstallError(
-                    "Variant +cxx requires the @master version of Julia")
+                    'Variant +cxx requires the @master version of Julia')
```

Comment:
This can be refactored into a conflict statement. Then it will fail during concretization instead of during installation.

Reply:
Actually, I should check whether newer released versions support that variant now.

Reply:
Yep, versions >= 1 can build with +cxx.